.github/workflows/ci.yml

```diff
@@ -9,16 +9,16 @@ jobs:
         name: Assemble
         runs-on: ubuntu-latest
         timeout-minutes: 25
-        env:
-            JAVA_TOOL_OPTIONS: -Xmx6g
-            GRADLE_OPTS: -Dorg.gradle.daemon=false -Dorg.gradle.workers.max=2 -Dkotlin.incremental=false -Dkotlin.compiler.execution.strategy=in-process
         steps:
-            - uses: actions/checkout@v2
-            - uses: gradle/wrapper-validation-action@v1
-            - uses: actions/setup-java@v1
+            - uses: actions/checkout@v4
+            - uses: gradle/wrapper-validation-action@v2
+            - uses: actions/setup-java@v4
               with:
-                  java-version: 11
-            - uses: actions/cache@v2
+                distribution: 'zulu'
+                java-version: '17'
+            - name: Setup Gradle
+              uses: gradle/actions/setup-gradle@v3
+            - uses: actions/cache@v4
```

Comment:
Remove `cache` step for Gradle, it is already set in `gradle/actions/setup-gradle`

Reply:
Good catch. will do!